syntax = "proto3";

package pb;

import "google/protobuf/any.proto";
import "google/protobuf/empty.proto";


service Event {

    rpc Reload(ReloadRequest) returns (CommonResponse) {};
    rpc Status(google.protobuf.Empty) returns (StatusResponse) {};
    rpc Stop(StopRquest) returns (CommonResponse) {};

    rpc EventTracker(EventTrackerRequest) returns (EventTrackerResponse) {};
    rpc AddNewUser(AddNewUserRequest) returns (AddNewUserResponse) {};
}

message CommonResponse {
  int32 code = 1;
  string description  = 2;
}

message ReloadRequest {
  google.protobuf.Any dicts = 1;
}

message StopRquest {
  optional int32 signal = 1;
}

message StatusResponse {
  int32 code = 1;
  google.protobuf.Any status = 2;
}

message EventTrackerRequest {
  string apiVersion = 1;
  string metadata = 2;
  string type = 3;
  google.protobuf.Any requestBody = 4;
}

message EventTrackerResponse {
  int32 code = 1;
  string description = 2;
}

message AddNewUserRequest {
  string apiVersion = 1;
  string metadata = 2;
  string type = 3;
  google.protobuf.Any requestBody = 4;
}

message AddNewUserResponse {
  int32 code = 1;
  string description = 2;
}
